dao/courses.go

```diff
@@ -214,19 +214,34 @@ func (d coursesDao) GetInvitedUsersForCourse(course *model.Course) error {
 	return DB.Preload("Users", "role = ?", model.GenericType).Find(course).Error
 }
 
-func (d coursesDao) GetCourseBySlugYearAndTerm(ctx context.Context, slug string, term string, year int) (model.Course, error) {
-	cachedCourses, found := Cache.Get(fmt.Sprintf("courseBySlugYearAndTerm%v%v%v", slug, term, year))
+// GetCourseBySlugYearAndTerm gets course by slug, year, term from database. Preloads personal lecture titles if userID is not 0
+func (d coursesDao) GetCourseBySlugYearAndTerm(ctx context.Context, slug string, term string, year int, userID uint) (model.Course, error) {
+	var cacheKey string
+	if userID == 0 {
+		cacheKey = fmt.Sprintf("courseBySlugYearAndTerm%v%v%v", slug, term, year)
+	} else {
+		cacheKey = fmt.Sprintf("courseBySlugYearTermUser%v%v%v%v", slug, term, year, userID)
```

Comment:
Not going to be an issue soon if we don't have weird years, but this introduces a cache key collision if e.g. year=1, and userId=11 we produce the same key as year=11, userId=1. I suggest adding a `-` in between